src/core_plugins/input_control_vis/public/components/editor/index_pattern_select.js

```diff
@@ -46,13 +46,15 @@ export class IndexPatternSelect extends Component {
       return;
     }
 
-    const indexPattern = await this.props.getIndexPattern(indexPatternId);
-
-    if (!this._isMounted) {
+    let indexPattern;
+    try {
+      indexPattern = await this.props.getIndexPattern(indexPatternId);
+    } catch (err) {
+      // index pattern no longer exists
       return;
     }
 
-    if (!indexPattern) {
+    if (!this._isMounted) {
       return;
```

Comment:
This looks really similar to the previous file (field_select.js) should they share some logic (e.g. the try / catch fetch index pattern, maybe _isMounted / _hasUnmounted checks? Dunno.